dotnet/src/Connectors/Connectors.OpenAI/Core/ClientCore.ChatCompletion.cs

```diff
@@ -1234,6 +1234,22 @@ private ToolCallingConfig GetToolCallingConfiguration(Kernel? kernel, OpenAIProm
             return new(tools, toolChoice, autoInvoke, maximumAutoInvokeAttempts);
         }
 
+        if (config.Choice == FunctionChoice.Required)
```

Comment:
Would now be a good time for your TODO above to consolidate the if statements?

Reply:
I plan to do it in a separate "improvments" PR later.